scripts/download_from_lesswrong.com.py

```diff
@@ -1,2 +1,130 @@
-import requests
-print("I am empty script")
+import sys
+import json
+import time
+import logging as log
+import lib.lesswrong as lw
+from pathlib import Path
+
+log.basicConfig(format="%(asctime)s %(levelname)s %(message)s", level=log.INFO)
+
+ROOT = "/tmp/lesswrong.com/"
+COLLECTION_SLUG = "rationality"
+
+###
+def safe_dirname(name: str) -> str:
+    return name.replace("/", "|")
+
+
+def write_post(postdir: str, post: dict):
+    metadata = {
+        "id": post["_id"],
+        "author": post["user"]["displayName"],
+        "modified_at": post["modifiedAt"],
+        "url": post["pageUrl"],
+    }
+
+    if "tags" in post and isinstance(post["tags"], list):
+        metadata["tags"] = []
+
+        for tag in post["tags"]:
+            metadata["tags"].append(tag["name"])
+
+    try:
+        log.info("Create data files for a post '%s'", postdir)
+
+        p = Path(postdir, "content.html")
+        p.write_text(post["contents"]["html"])
+
+        p = Path(postdir, "metadata")
+        p.write_text(json.dumps(metadata))
+    except Exception as err:
+        log.critical("Can't create data files for post '%s': %s", postdir, err)
+        sys.exit(err)
+
+    return
+
+
+def main():
+    log.info("Download collection '%s'", COLLECTION_SLUG)
+    collection = lw.download_collection(COLLECTION_SLUG)
+
+    if "error" in collection:
+        log.critical(
+            "Collection '%s' not downloaded: %s", COLLECTION_SLUG, collection["error"]
+        )
+        sys.exit("Collection download error")
+
+    collection = collection["data"]
+
+    p_base = Path(ROOT, collection["title"])
+
+    posts_for_download = {}
+
+    log.debug("Create directories tree at '%s'", p_base.absolute())
+
+    for book_i, book in enumerate(collection["books"]):
+        book_dirname = str(book_i + 1).zfill(2) + "_" + safe_dirname(book["title"])
+        book_p = Path(p_base.absolute(), book_dirname)
+
+        for sequence_i, sequence in enumerate(book["sequences"]):
+            sequence_dirname = (
+                str(sequence_i + 1).zfill(2) + "_" + safe_dirname(sequence["title"])
+            )
+            sequence_p = Path(book_p.absolute(), sequence_dirname)
+
+            post_i = 1
+            if "contents" in sequence and isinstance(sequence["contents"], dict):
+
+                intro_post = {
+                    "title": "Intro",
+                    "contents": {"html": sequence["contents"]["html"]},
+                    "modifiedAt": sequence["contents"]["editedAt"],
+                    "_id": sequence["_id"],
+                    "user": sequence["user"],
+                    "pageUrl": "https://www.lesswrong.com/s/" + sequence["_id"],
```

Comment:
Do they have info about authors? Or it's in body?

Reply:
It's "user"

Reply:
Wondering do they have cases when it's more than one person

Reply:
i’ll look later